yard-impl2/src/main/java/org/kie/yard/impl2/LiteralExpressionInterpreter.java

```diff
@@ -0,0 +1,55 @@
+package org.kie.yard.impl2;
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.script.Bindings;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+public class LiteralExpressionInterpreter implements Firable {
+    private final String name;
+    private final QuotedExprParsed quoted;
+    private final ScriptEngine engine;
+    private final CompiledScript compiledScript;
+
+    public LiteralExpressionInterpreter(String nameString, QuotedExprParsed quotedExprParsed) {
+        this.name = nameString;
+        this.quoted = quotedExprParsed;
+        try {
+            ScriptEngineManager manager = new ScriptEngineManager();
+            engine = manager.getEngineByName("jshell");
+            Compilable compiler = (Compilable) engine;
+            compiledScript = compiler.compile(quoted.getRewrittenExpression());
```

Comment:
Two points:

1) I would suggest to move all this logic away into the YaRDParser - or into a factory for these components, if we want to separate concerns. My preference is for constructors where you pass just a list of objects that are associated to instance fields. In this case the signature for the method would be  LiteralExpressionInterpreter(String nameString, QuotedExprParsed quotedExprParsed, ScriptEngine engine, CompiledScript compiledScript)
2) the name is kind of strange. This is just an expression with free variables, that are bind at the evaluation moment. Perhaps somehting like LiteralExpressionWithFreeVariables?

Reply:
as we progress on the different implementations (ie impl1 "vs" impl2) indeed it should emerge the best way to refactor this